src/Internal/DefaultCacheProvider.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tebru\Gson\Internal;
+
+use Psr\SimpleCache\CacheInterface;
+use Symfony\Component\Cache\Adapter\Psr16Adapter;
+
+final class DefaultCacheProvider
```

Comment:
You can just call this `CacheProvider` since it doesn't implement an interface, there won't be multiple versions.